addons/mail/models/mail_channel.py

```diff
@@ -239,6 +239,10 @@ def _unlink_except_all_employee_channel(self):
             raise UserError(_('You cannot delete those groups, as the Whole Company group is required by other modules.'))
 
     def write(self, vals):
+        if 'channel_type' in vals:
+            failing_channels = self.sudo().filtered(lambda channel: channel.channel_type is not None and channel.channel_type != vals.get('channel_type'))
```

Comment:
We should merge PR about `channel_type` being required first, and remove condition on "is not None" from here.
Also "is not None" is probably not covering the expected cases anyway, as the ORM has a tendency to have `False` as as empty value, even for non-boolean fields.

Reply:
ok, and you can have a look at the 'required' task. it is ready to be reviewed!